iroh/src/util/collection.rs

```diff
@@ -202,6 +232,35 @@ impl Collection {
         Ok(links_tag)
     }
 
+    /// Store a collection in a store. returns the root hash of the collection
+    /// as a TempTag.
+    pub async fn store_iroh<C>(
+        self,
+        blobs: &crate::client::blobs::Client<C>,
+        tag: SetTagOption,
+    ) -> anyhow::Result<(Hash, Tag)>
+    where
+        C: ServiceConnection<RpcService>,
+    {
+        let (links, meta) = self.into_parts();
+        let meta_bytes = postcard::to_stdvec(&meta)?;
+        let res_meta = blobs.add_bytes(meta_bytes).await?;
+
+        let links_bytes = std::iter::once(res_meta.hash)
+            .chain(links)
+            .collect::<HashSeq>();
+
+        // delete meta_tag
+        let tags = crate::client::tags::Client {
+            rpc: blobs.rpc.clone(),
+        };
+        tags.delete(res_meta.tag).await?;
```

Comment:
Shouldn't this happen after the hashseq is added? If not, what's protecting the meta_bytes if a GC would happen between here and the add_stream_hash_seq call?

Reply:
fixed